src/v/redpanda/tests/debug_bundle_tests.cc

```diff
@@ -0,0 +1,295 @@
+                return true;
+            }
+
+            if (is_params_flag(flag)) {
+                vlog(test_log.info, "Found param {}", flag);
+                return true;
+            }
+
+            if (is_cred_flag(flag)) {
+                vlog(test_log.info, "Found cred {}", flag);
+                return true;
+            }
+
+            return false;
+        };
+
+        std::optional<debug_bundle_credentials> creds_opt = std::make_optional(
+          creds);
+        auto rpk_argv = detail::make_rpk_args(
+          rpk_path, full_file_path, params, creds_opt);
+        for (const auto& flag : rpk_argv) {
+            BOOST_CHECK(is_flag_ok(flag));
+        }
+    }
+}
+
+SEASTAR_THREAD_TEST_CASE(check_process_spawn) {
+    const std::filesystem::path true_cmd{"/bin/true"}, false_cmd{"/bin/false"},
+      sleep_cmd{"/bin/sleep"};
+
+    // Check failed cmd
+    {
+        auto process = ss::experimental::spawn_process(false_cmd).get();
+        auto wstatus = process.wait().get();
+        // Expect to see an error log with exit status 1
+        auto err = detail::handle_wait_status(
+          wstatus, false_cmd.string(), default_bundle_name());
+        BOOST_CHECK_EQUAL(
+          detail::make_rpk_err_value(err),
+          rpk::debug_bundle_error::process_fail);
+    }
+
+    // Check sucessful cmd
+    {
+        auto process = ss::experimental::spawn_process(true_cmd).get();
+        auto wstatus = process.wait().get();
+        auto err = detail::handle_wait_status(
+          wstatus, true_cmd.string(), default_bundle_name());
+        BOOST_CHECK_EQUAL(
+          detail::make_rpk_err_value(err), rpk::debug_bundle_error::success);
+    }
+
+    std::vector<ss::sstring> sleep_argv{sleep_cmd.string(), "30"};
+
+    // Check SIGTERM'd cmd
+    {
+        auto process = ss::experimental::spawn_process(
+                         sleep_cmd, {.argv = sleep_argv})
+                         .get();
+        process.terminate();
+        auto wstatus = process.wait().get();
+        // Expect to see an error log with signal 15
+        auto err = detail::handle_wait_status(
+          wstatus, sleep_cmd.string(), default_bundle_name());
+        BOOST_CHECK_EQUAL(
+          detail::make_rpk_err_value(err),
+          rpk::debug_bundle_error::process_fail);
+    }
+
+    // Check SIGKILL'd cmd
+    {
+        std::vector<ss::sstring> sleep_argv{sleep_cmd.string(), "30"};
+        auto process = ss::experimental::spawn_process(
+                         sleep_cmd, {.argv = sleep_argv})
+                         .get();
+        process.kill();
+        auto wstatus = process.wait().get();
+        // Expect to see an error log with signal 9
+        auto err = detail::handle_wait_status(
+          wstatus, sleep_cmd.string(), default_bundle_name());
+        BOOST_CHECK_EQUAL(
+          detail::make_rpk_err_value(err),
+          rpk::debug_bundle_error::process_fail);
+    }
+}
```

Comment:
These tests aren't necessary.  These are just testing `ss::process`.

Reply:
These tests are to check `detail::handle_wait_status`. I'll change the name to be more clear.

Reply:
You probably don't need to fire up a `ss::process` to test it.  You could just craft the appropriate `wait_status` and test `detail::handle_wait_status` that way.